app/helpers/assessment_handin_core.rb

```diff
@@ -90,11 +90,17 @@ def saveHandin(sub, app_id = nil)
     end
 
     submissions = []
+
+    # group_key = group_name_submitter_email_handin_filename_timestamp
+    group_key = "#{group.name}_#{@cud.user.email}_#{@assessment.handin_filename}_"
+    group_key += Time.current.utc.to_s(:number)
+
```

Comment:
just to clarify, this is to uniquely identify the group submission, and this is needed because there might be a case where it is like version 1 for group member a, and version 2 for group member b?

Reply:
Might there be an issue if any of the fields contain an underscore?

Reply:
Yes, that seems to be precisely the case, especially if one student made submissions before joining the group, so the version numbers might differ. This allows us to figure out exactly which "submissions" are part of a certain group submission.

Using underscores as a separator seems a bit iffy to me, but a name collision seems pretty much impossible especially given the timestamp at the end, so this is fine.